processor/redactionprocessor/internal/url/sanitizer.go

```diff
@@ -0,0 +1,128 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package url // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/redactionprocessor/internal/url"
+
+import (
+	"embed"
+	"encoding/json"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/AlessandroPomponio/go-gibberish/gibberish"
+	"github.com/AlessandroPomponio/go-gibberish/structs"
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+//go:embed classifier.json
+var classifierData embed.FS
+
+type URLSanitizer struct {
+	maxSegments int
+	words       *lru.Cache[string, bool]
+	classifier  *structs.GibberishData
+	replaceWith string
+	attributes  []string
+}
+
+func NewURLSanitizer(config URLSanitizationConfig) (*URLSanitizer, error) {
+	if config.Replacement == "" {
+		config.Replacement = "*"
+	}
+
+	if config.MaxSegments == 0 {
+		config.MaxSegments = 10
+	}
+
+	if config.CacheSize == 0 {
+		config.CacheSize = 8192
+	}
+
+	content, err := classifierData.ReadFile("classifier.json")
+	if err != nil {
+		return nil, fmt.Errorf("unable to read knowledge base content: %w", err)
+	}
+
+	var classifier structs.GibberishData
+	err = json.Unmarshal(content, &classifier)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshal knowledge base content: %w", err)
+	}
+
+	words, err := lru.New[string, bool](config.CacheSize)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create LRU cache: %w", err)
+	}
+
+	return &URLSanitizer{
+		maxSegments: config.MaxSegments,
+		words:       words,
+		classifier:  &classifier,
+		replaceWith: config.Replacement,
+		attributes:  config.Attributes,
+	}, nil
+}
+
+func (s *URLSanitizer) SanitizeAttributeURL(url, attributeKey string) string {
+	if !slices.Contains(s.attributes, attributeKey) {
+		return url
+	}
+
+	return s.SanitizeURL(url)
+}
+
+// SanitizeURL sanitizes the given URL by removing any gibberish words.
+// https://github.com/open-telemetry/opentelemetry-ebpf-instrumentation/blob/38ca7938595409b8ffe6b897c14a0e3280dd2941/pkg/components/transform/route/cluster.go#L48
+func (s *URLSanitizer) SanitizeURL(url string) string {
+	if url == "" {
+		return url
+	}
+
+	segments := strings.Split(url, "/")
```

Comment:
It might be good to make a benchmark to compare the performance of the cluster.go code vs this implementation. When I initially wrote the code I used a similar approach to the one implemented here, but the overhead was unacceptable for our project, since this runs on every HTTP span.

Reply:
So there's quite a bit of difference in both CPU time and memory allocations between the two, which one was which?

Reply:
Second implementation it's what there is now in the PR. Totally based on the library you implemented.